packages/use-intl/src/core/utils/TimeZones.tsx

```diff
@@ -0,0 +1,422 @@
+	| "Asia/Vladivostok"
+	| "Asia/Yakutsk"
+	| "Asia/Yangon"
+	| "Asia/Yekaterinburg"
+	| "Asia/Yerevan"
+	| "Atlantic/Azores"
+	| "Atlantic/Bermuda"
+	| "Atlantic/Canary"
+	| "Atlantic/Cape_Verde"
+	| "Atlantic/Faroe"
+	| "Atlantic/Madeira"
+	| "Atlantic/Reykjavik"
+	| "Atlantic/South_Georgia"
+	| "Atlantic/St_Helena"
+	| "Atlantic/Stanley"
+	| "Australia/Adelaide"
+	| "Australia/Brisbane"
+	| "Australia/Broken_Hill"
+	| "Australia/Currie"
+	| "Australia/Darwin"
+	| "Australia/Eucla"
+	| "Australia/Hobart"
+	| "Australia/Lindeman"
+	| "Australia/Lord_Howe"
+	| "Australia/Melbourne"
+	| "Australia/Perth"
+	| "Australia/Sydney"
+	| "Europe/Amsterdam"
+	| "Europe/Andorra"
+	| "Europe/Astrakhan"
+	| "Europe/Athens"
+	| "Europe/Belgrade"
+	| "Europe/Berlin"
+	| "Europe/Bratislava"
+	| "Europe/Brussels"
+	| "Europe/Bucharest"
+	| "Europe/Budapest"
+	| "Europe/Busingen"
+	| "Europe/Chisinau"
+	| "Europe/Copenhagen"
+	| "Europe/Dublin"
+	| "Europe/Gibraltar"
+	| "Europe/Guernsey"
+	| "Europe/Helsinki"
+	| "Europe/Isle_of_Man"
+	| "Europe/Istanbul"
+	| "Europe/Jersey"
+	| "Europe/Kaliningrad"
+	| "Europe/Kiev"
+	| "Europe/Kirov"
+	| "Europe/Lisbon"
+	| "Europe/Ljubljana"
+	| "Europe/London"
+	| "Europe/Luxembourg"
+	| "Europe/Madrid"
+	| "Europe/Malta"
+	| "Europe/Mariehamn"
+	| "Europe/Minsk"
+	| "Europe/Monaco"
+	| "Europe/Moscow"
+	| "Europe/Oslo"
+	| "Europe/Paris"
+	| "Europe/Podgorica"
+	| "Europe/Prague"
+	| "Europe/Riga"
+	| "Europe/Rome"
+	| "Europe/Samara"
+	| "Europe/San_Marino"
+	| "Europe/Sarajevo"
+	| "Europe/Saratov"
+	| "Europe/Simferopol"
+	| "Europe/Skopje"
+	| "Europe/Sofia"
+	| "Europe/Stockholm"
+	| "Europe/Tallinn"
+	| "Europe/Tirane"
+	| "Europe/Ulyanovsk"
+	| "Europe/Uzhgorod"
+	| "Europe/Vaduz"
+	| "Europe/Vatican"
+	| "Europe/Vienna"
+	| "Europe/Vilnius"
+	| "Europe/Volgograd"
+	| "Europe/Warsaw"
+	| "Europe/Zagreb"
+	| "Europe/Zaporozhye"
+	| "Europe/Zurich"
+	| "Indian/Antananarivo"
+	| "Indian/Chagos"
+	| "Indian/Christmas"
+	| "Indian/Cocos"
+	| "Indian/Comoro"
+	| "Indian/Kerguelen"
+	| "Indian/Mahe"
+	| "Indian/Maldives"
+	| "Indian/Mauritius"
+	| "Indian/Mayotte"
+	| "Indian/Reunion"
+	| "Pacific/Apia"
+	| "Pacific/Auckland"
+	| "Pacific/Bougainville"
+	| "Pacific/Chatham"
+	| "Pacific/Chuuk"
+	| "Pacific/Easter"
+	| "Pacific/Efate"
+	| "Pacific/Enderbury"
+	| "Pacific/Fakaofo"
+	| "Pacific/Fiji"
+	| "Pacific/Funafuti"
+	| "Pacific/Galapagos"
+	| "Pacific/Gambier"
+	| "Pacific/Guadalcanal"
+	| "Pacific/Guam"
+	| "Pacific/Honolulu"
+	| "Pacific/Kiritimati"
+	| "Pacific/Kosrae"
+	| "Pacific/Kwajalein"
+	| "Pacific/Majuro"
+	| "Pacific/Marquesas"
+	| "Pacific/Nauru"
+	| "Pacific/Niue"
+	| "Pacific/Norfolk"
+	| "Pacific/Noumea"
+	| "Pacific/Pago_Pago"
+	| "Pacific/Palau"
+	| "Pacific/Pitcairn"
+	| "Pacific/Pohnpei"
+	| "Pacific/Port_Moresby"
+	| "Pacific/Rarotonga"
+	| "Pacific/Tahiti"
+	| "Pacific/Tarawa"
+	| "Pacific/Tongatapu"
+	| "Pacific/Wake"
+	| "Pacific/Wallis"
+	| "UTC"
+	| "W-SU"
+	| "WET"
+	| "Zulu";
```

Comment:
I'm wondering if we should allow unknown strings here too (see [this trick](https://stackoverflow.com/a/74467583/343045)).

The benefit of allowing unknown strings is that we don't have to publish a new version in case our list gets outdated. This might even be critical for users on outdated versions, if we've released breaking changes in the meantime, otherwise we'd have to backport changed time zones. AFAIK React does this too, e.g. for ARIA roles.

The downside is of course that we don't fail for invalid time zones.

However, JavaScript runtimes throw an error for unknown time zones:

<img width="430" alt="Screenshot 2023-06-28 at 17 09 31" src="https://github.com/amannn/next-intl/assets/4038316/fcb57142-8dd5-4a41-9553-496c789612d8">

(I've supplied "Unknown" in this example)

Therefore, I'd say we should support unknown strings.